src/pybind/mgr/snap_schedule/fs/schedule_client.py

```diff
@@ -46,8 +49,8 @@ def open_ioctx(self, pool):
         raise
 
 
-def updates_schedule_db(func):
-    def f(self, fs, schedule_or_path, *args):
+def updates_schedule_db(func: Callable[..., None]) -> Callable[..., None]:
+    def f(self: 'SnapSchedClient', fs: str, schedule_or_path: str, *args: Any) -> None:
```

Comment:
yeah, what i didn't like was `cast(FuncT, f)`. because this "decorator" actually changes the return type (again!). i will change it so it actually return whatever `func` returns. this should be fine in this case, as all the decorated functions return `None`.